src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/create-cluster/create-cluster.component.ts

```diff
@@ -140,12 +157,37 @@ export class CreateClusterComponent implements OnDestroy {
           }
         });
     }
+
+    if (this.deploymentOption) {
+      const trackingId = `${this.deploymentOption} optmized deployment`;
+      this.taskWrapper
+        .wrapTaskAroundCall({
+          task: new FinishedTask('osd/' + URLVerbs.CREATE, {
+            tracking_id: trackingId
+          }),
+          call: this.osdService.create(
+            [{ option: this.deploymentOption }],
+            trackingId,
+            'predefined'
+          )
+        })
+        .subscribe({
+          error: (error) => error.preventDefault(),
+          complete: () => {
+            this.submitAction.emit();
+          }
+        });
+    }
   }
 
   getDriveGroup(driveGroup: DriveGroup) {
     this.driveGroup = driveGroup;
   }
 
+  getDeploymentOption(option: string) {
+    this.deploymentOption = option;
+  }
```

Comment:
This is a brainwrecker... Because it says `get*` but it's actually setting something. We shouldn't name things in a misleading manner. This applies also to the above `getDriveGroup` one...

If we want a getter/accessor method, then [this proper typescript form is](https://www.typescriptlang.org/docs/handbook/2/classes.html#getters--setters):
```suggestion
  get deploymentOption(): DeploymentOptionEnum {
    return this._deploymentOption;
  }
  set deploymentOption(option: DeploymentOptionEnum):  {
    this._deploymentOption = option;
  }
```

Reply:
Done! thanks